src/buildstream/sandbox/_sandboxbuildboxrun.py

```diff
@@ -148,7 +173,14 @@ def _execute_action(self, action, flags):
                 interactive=(flags & _SandboxFlags.INTERACTIVE),
             )
 
-            return remote_execution_pb2.ActionResult().FromString(result_file.read())
+            action_result = remote_execution_pb2.ActionResult().FromString(result_file.read())
+
+        if self.re_remote and context.remote_execution_specs.storage_spec and context.remote_cache_spec:
+            # This ensures that the outputs are uploaded to the cache storage-service
+            # in case different CAS remotes have been configured in the `cache` and `remote-execution` sections.
```

Comment:
I'm having a hard time wrapping my head around this one. Since we're building locally, shouldn't the outputs be uploaded to the cache storage-service by default? Then we only need to upload them to the remote-execution storage if we're also uploading the action?

Or is it due to the fact that buildbox-run can only take a single instance for both the nested REAPI and where it stores its output? So it will upload its output to the remote-execution CAS as well and we need to transfer it to the cache CAS?

I agree this is an exotic configuration, and shouldn't block landing this MR. I just wanted to raise it just in case.